src/openFindOrder.php

```diff
@@ -392,6 +394,54 @@ private function send_error($message, $response_tag = 'findOrdersResponse', $deb
 
     return $response;
   }
+
+  /**
+   * @param $param
+   * @throws Exception
+   *
+   * AuditTrail:
+   *   string $accessingUser The login token retrieved from Adgangsplatformen.
+   *   array $clientIps The client ip addresses and perhaps a proxy etc.
+   *   string $appName The application that calls this method, Bobbasen, Netpunkt, BibliotekDK etc.
+   *   string $action Are we writing or reading to/from the system.
+   *   string $owningUser The owner of the data being watched or written.
+   *   array $accessInfo The data that are being handled for the request.
+   */
+  private function auditTrail($param) {
+    $user = $param->authentication->_value->groupIdAut->_value . '::' . $param->authentication->_value->userIdAut->_value;
+    try {
+      \DBC\AT\AuditTrail::log(
+          $user == '::' ? 'no-user-specified' : $user,
+          [$_SERVER['HTTP_X_FORWARDED_FOR'] ?: $_SERVER['REMOTE_ADDR']],
+          'openFindOrder' . '::' . $this->soap_action,
+          'read',
+          $param->requesterAgencyId->_value,
+          [json_encode(self::cleanBadgerfish($param))]
+      );
+    } catch (Exception $e) {
+      VerboseJson::log(ERROR, 'Cannot write audit trail. Message: ' . $e->getMessage());
+    }
+  }
+
+  /** remove namespace objects and implode value level
+   *
+   * @param $misc
+   * @return bool|float|int|stdClass|string
+   */
+  private function cleanBadgerfish($misc) {
+    if (is_scalar($misc)) return $misc;
+    $ret = new stdClass();
+    foreach ($misc as $key => $value) {
+      if ($key !== '_namespace') {
```

Comment:
Er den her if nødvendig? Du spørger nedenunder om if er lig '_value', og du gør intet, hvis $key er '_namespace'.

Reply:
_namespace elementerne skal smides væk, så det er korrekt at der intet gøres

Reply:
jeg ville nok ha' lavet det i en switch (for at undgå nestede if'er), men det her er også ok